src/game_api/movable_behavior.cpp

```diff
@@ -190,36 +194,213 @@ void update_movable(Movable* movable, bool disable_gravity)
+}
+void reset_clingy_ents()
+{
+    for (const ClimbableEntityRecoveryInfo& recovery_info : g_temporary_climbable_entities)
+    {
+        if (recovery_info.climbable)
+        {
+            recovery_info.entity->type->properties_flags |= 0x20;
+        }
+        else
+        {
+            recovery_info.entity->type->properties_flags &= ~0x20;
+        }
+    }
+    g_temporary_climbable_entities.clear();
+}
+
+using LedgeGrabForceState = bool(MovableBehavior*, Movable*);
+LedgeGrabForceState* g_original_ledge_grab_force_state{nullptr};
+bool ledge_grab_force_state(MovableBehavior* behavior, Movable* player)
+{
+    g_expecting_ledge_hang_test = true;
+    ON_SCOPE_EXIT(g_expecting_ledge_hang_test = false; reset_clingy_ents());
+    return g_original_ledge_grab_force_state(behavior, player);
+}
+using LedgeGrabOnEnter = bool(MovableBehavior*, Movable*);
+LedgeGrabOnEnter* g_original_ledge_grab_on_enter{nullptr};
+void ledge_grab_on_enter(MovableBehavior* behavior, Movable* player)
+{
+    g_expecting_ledge_hang_test = true;
+    ON_SCOPE_EXIT(g_expecting_ledge_hang_test = false; reset_clingy_ents());
+    g_original_ledge_grab_on_enter(behavior, player);
+}
+using LedgeGrabGetNextStateId = uint8_t(MovableBehavior*, Movable*);
+LedgeGrabGetNextStateId* g_original_ledge_grab_get_next_state_id{nullptr};
+uint8_t ledge_grab_get_next_state_id(MovableBehavior* behavior, Movable* player)
+{
+    g_expecting_ledge_hang_test = true;
+    ON_SCOPE_EXIT(g_expecting_ledge_hang_test = false; reset_clingy_ents());
+    return g_original_ledge_grab_get_next_state_id(behavior, player);
+}
+
+using EntityTurn = void(Entity*, bool);
+EntityTurn* g_entity_turn_trampoline{nullptr};
+void entity_turn(Entity* self, bool apply)
+{
+    auto find_entity_pred = [=](const TurningEntityInfo& turning_info)
+    { return turning_info.entity == self; };
+
+    if (std::find_if(g_entities_disabled_turning.begin(), g_entities_disabled_turning.end(), find_entity_pred) == g_entities_disabled_turning.end())
+    {
+        g_entity_turn_trampoline(self, apply);
+    }
+}
+
+#define HOOK_MOVE_ENTITY
 #ifdef HOOK_MOVE_ENTITY
 UpdateMovable* g_update_movable_trampoline{nullptr};
 void update_movable(Movable& movable, const Vec2& move_xy, float sprint_factor, bool apply_move, bool disable_gravity, bool on_ladder, bool param_7)
 {
     g_update_movable_trampoline(movable, move_xy, sprint_factor, apply_move, disable_gravity, on_ladder, param_7);
 }
-
-#include <detours.h>
+#endif
 
 void init_behavior_hooks()
 {
     DetourTransactionBegin();
     DetourUpdateThread(GetCurrentThread());
 
     auto memory = Memory::get();
-    g_update_movable_trampoline = (UpdateMovable*)memory.at_exe(0x228e3580);
 
+    g_get_entity_close_to_trampoline = (GetEntityCloseTo*)get_address("layer_get_entity_close_to"sv);
+    DetourAttach((void**)&g_get_entity_close_to_trampoline, (GetEntityCloseTo*)get_entity_close_to);
+
+    g_entity_turn_trampoline = (EntityTurn*)get_address("entity_turn"sv);
```

Comment:
could be replaced with `get_virtual_function_address` to not add another pattern to search for at the start

Reply:
Right, you mentioned before, I forgot about that